website/project/model.py

```diff
@@ -3479,6 +3419,204 @@ def require_approval(self, user, notify_initiator_on_complete=False):
             save=True,
         )
 
+    is_institution = fields.BooleanField(default=False, index=True)
+
+    institution_id = fields.StringField(unique=True)
+    institution_domain = fields.StringField(list=True)
+    institution_auth_url = fields.StringField(validate=URLValidator())
+    institution_logo_name = fields.StringField()
+
+    @classmethod
+    def find(cls, query=None, allow_institution=False, **kwargs):
+        if not allow_institution:
+            query = (query & Q('is_institution', 'ne', True)) if query else Q('is_institution', 'ne', True)
+        return super(Node, cls).find(query, **kwargs)
+
+    @classmethod
+    def find_one(cls, query=None, allow_institution=False, **kwargs):
+        if not allow_institution:
+            query = (query & Q('is_institution', 'ne', True)) if query else Q('is_institution', 'ne', True)
+        return super(Node, cls).find_one(query, **kwargs)
+
+    @classmethod
+    def find_by_institution(cls, inst, query=None):
+        inst_node = inst.node
+        query = query & Q('_primary_institution', 'eq', inst_node)
+        return cls.find(query, allow_institution=True)
+
+    # Primary institution node is attached to
+    _primary_institution = fields.ForeignField('node')
+
+    @property
+    def primary_institution(self):
+        return Institution(self._primary_institution) if self._primary_institution else None
+
+    @primary_institution.setter
+    def primary_institution(self, institution):
+        self._primary_institution = institution.node if institution else None
+
+    _affiliated_institutions = fields.ForeignField('node', list=True)
+
+    @property
+    def affiliated_institutions(self):
+        return SpecialList(self, '_affiliated_institutions', [Institution(node) for node in self._affiliated_institutions])
+
+    def add_primary_institution(self, user, inst, log=True):
+        if not user.is_affiliated_with_institution(inst):
+            raise UserNotAffiliatedError('User is not affiliated with {}'.format(inst.name))
+        if inst.node == self.primary_institution:
+            return False
+        previous = Institution(self.primary_institution) if self.primary_institution else None
+        self.primary_institution = inst
+        if inst.node not in self.affiliated_institutions:
```

Comment:
Similar again, should this be `if inst not in self.affiliated_institutions` since `self.affiliated_institutions` is a list of `Institutions`?